deebotozmo/cli.py

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/env python3
+"""Cli module."""
+import asyncio
+import base64
+import configparser
+import itertools
+import json
+import logging
+import os
+import platform
+import sys
+import time
+from dataclasses import asdict
+from functools import wraps
+from typing import Any, Callable, Tuple
+
+import aiohttp
+import click
+
+from deebotozmo.commands import (
+    Charge,
+    CleanCustomArea,
+    CleanPause,
+    CleanResume,
+    CleanSpotArea,
+    CleanStart,
+    PlaySound,
+    SetFanSpeed,
+    SetWaterLevel,
+)
+from deebotozmo.ecovacs_api import EcovacsAPI
+from deebotozmo.ecovacs_mqtt import EcovacsMqtt
+from deebotozmo.events import (
+    BatteryEvent,
+    CleanLogEvent,
+    FanSpeedEvent,
+    MapEvent,
+    RoomsEvent,
+    StatsEvent,
+    WaterInfoEvent,
+)
+from deebotozmo.util import md5
+from deebotozmo.vacuum_bot import VacuumBot
+
+
+def coro(func: Callable) -> Callable:
+    """Wrap around a function to run it in coroutine."""
+
+    @wraps(func)
+    def wrapper(*args: Tuple, **kwargs: Tuple) -> Any:
+        loop = asyncio.get_event_loop()
+        return loop.run_until_complete(func(*args, **kwargs))
+
+    return wrapper
+
+
+def config_file() -> str:
+    """Return the path to config file."""
+    if platform.system() == "Windows" and "APPDATA" in os.environ:
+        return os.path.join(str(os.getenv("APPDATA")), "deebotozmo.conf")
+    return os.path.expanduser("~/.config/deebotozmo.conf")
+
+
+def config_file_exists() -> bool:
+    """Check if the config file exists."""
+    return os.path.isfile(config_file())
+
+
+def read_config() -> configparser.SectionProxy:
+    """Read and parse the config file."""
+    parser = configparser.ConfigParser()
+    with open(config_file(), encoding="utf-8") as file:
+        parser.read_file(itertools.chain(["[global]"], file), source=config_file())
+    return parser["global"]
+
+
+def write_config(config: dict) -> None:
+    """Create a new config file."""
+    os.makedirs(os.path.dirname(config_file()), exist_ok=True)
+    with open(config_file(), "w", encoding="utf-8") as file:
```

Comment:
This isn't an issue because Python uses universal newlines by default. So it will treat `\n`, `\r\n`, and `\r` as a newline on all platforms, regardless if it is Windows. I tested this now by converting the config's line endings to `\r` and the script had no issue reading it 

I don't want to use ConfigParser's write functionality because it wouldn't work due to how we made the config. Our config right now has no section and ConfigParser needs a section.

Reply:
Instead of "sneaking" a global section in on reading you can write it directly in the file.
See my suggestions